src/python/tod/polyfilter.py

```diff
@@ -0,0 +1,137 @@
+           based on the detector flags.
+        poly_flag_mask (byte):  Bitmask to use when adding flags based
+           on polynomial filter failures.
+    """
+
+    def __init__(self, order=1, pattern='.*', name='signal',
+                 common_flag_name='common_flags', common_flag_mask=255,
+                 flag_name='flags', flag_mask=255, poly_flag_mask=1):
+
+        self._order = order
+        self._pattern = pattern
+        self._name = name
+        self._common_flag_name = common_flag_name
+        self._common_flag_mask = common_flag_mask
+        self._flag_name = flag_name
+        self._flag_mask = flag_mask
+        self._poly_flag_mask = poly_flag_mask
+
+        # We call the parent class constructor, which currently does nothing
+        super().__init__()
+
+    def exec(self, data):
+        """
+        Apply the polynomial filter to the signal.
+
+        Args:
+            data (toast.Data): The distributed data.
+        """
+        # the two-level pytoast communicator
+        comm = data.comm
+        # the global communicator
+        cworld = comm.comm_world
+        # the communicator within the group
+        cgroup = comm.comm_group
+        # the communicator with all processes with
+        # the same rank within their group
+        crank = comm.comm_rank
+
+        for obs in data.obs:
+            tod = obs['tod']
+            tod_first = tod.local_samples[0]
+
+             # Cache the output common flags
+            cachename = self._common_flag_name
+            if tod.cache.exists(cachename):
+                common_ref = tod.cache.reference(cachename)
+            else:
+                common_flag = tod.read_common_flags()
+                common_ref = tod.cache.put(cachename, common_flag)
+                del common_flag
+
+            for det in tod.local_dets:
+                # Test the detector pattern
+
+                # Cache the output signal
+                cachename = '{}_{}'.format(self._name, det)
+                if tod.cache.exists(cachename):
+                    ref = tod.cache.reference(cachename)
+                else:
+                    signal = tod.read(detector=det)
+                    ref = tod.cache.put(cachename, signal)
+
+                # Cache the output flags
+                cachename = '{}_{}'.format(self._flag_name, det)
+                if tod.cache.exists(cachename):
+                    flag_ref = tod.cache.reference(cachename)
+                else:
+                    # read_flags always returns both common and detector
+                    # flags but we already cached the common flags.
+                    flag, dummy = tod.read_flags(detector=det)
+                    flag_ref = tod.cache.put(cachename, flag)
+                    del flag, dummy
+
+                # Iterate over each chunk.
+
+                chunk_first = tod_first
+                for curchunk in range(tod.local_chunks[1]):
```

Comment:
This operator uses the data distribution chunks (stationary intervals) to specify the L/R scans, and then flags the turnarounds.  We could also use the "Interval" list to specify the good data regions.  If we use the data distribution chunks to specify the subscans, then they will be very small, which will limit the size of timestream over which noise can be simulated.  If we don't want to change this, then we need some other way of specifying the noise stationary period.